src/main/java/core/basesyntax/io/impl/CsvReportSerializer.java

```
@@ -0,0 +1,39 @@
+package core.basesyntax.io.impl;
+
+import core.basesyntax.io.ReportSerializer;
+import core.basesyntax.model.FruitType;
+import core.basesyntax.model.Report;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CsvReportSerializer implements ReportSerializer {
+    @Override
+    public void save(Report report, Path path) {
```

The `save` method doesn't validate its parameters. The program should validate inputs and fail fast with an informative message when given bad parameters (e.g., null report or null path). The project description explicitly asks to "try to anticipate and handle all invalid input data in advance". Add checks such as `Objects.requireNonNull(report, "Report must not be null")` and `Objects.requireNonNull(path, "Path must not be null")` and throw informative RuntimeException if validation fails. Reference: description: "Remember that your code will need to be tested, so try to anticipate and handle all invalid input data in advance."